Build failure: python312.pkgs.gensim

### Steps To Reproduce

Steps to reproduce the behavior:
1. nix-build -A python312.pkgs.gensim

### Build log

```
gensim/models/word2vec_inner.c: In function ‘__Pyx_PyInt_As_PY_LONG_LONG’:
gensim/models/word2vec_inner.c:12812:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
12812 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                     ^~
gensim/models/word2vec_inner.c:12867:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
12867 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                     ^~
gensim/models/word2vec_inner.c: In function ‘__Pyx_PyInt_As_int’:
gensim/models/word2vec_inner.c:13046:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
13046 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                     ^~
gensim/models/word2vec_inner.c:13101:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
13101 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                     ^~
gensim/models/word2vec_inner.c: In function ‘__Pyx_PyInt_As_unsigned_PY_LONG_LONG’:
gensim/models/word2vec_inner.c:13242:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
13242 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                     ^~
gensim/models/word2vec_inner.c:13297:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
13297 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                     ^~
gensim/models/word2vec_inner.c: In function ‘__Pyx_PyInt_As_npy_uint32’:
gensim/models/word2vec_inner.c:13438:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
13438 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                     ^~
gensim/models/word2vec_inner.c:13493:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
13493 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                     ^~
gensim/models/word2vec_inner.c: In function ‘__Pyx_PyInt_As_long’:
gensim/models/word2vec_inner.c:13634:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
13634 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                     ^~
gensim/models/word2vec_inner.c:13689:53: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
13689 |             const digit* digits = ((PyLongObject*)x)->ob_digit;
      |                                                     ^~
gensim/models/word2vec_inner.c: In function ‘__Pyx_PyIndex_AsSsize_t’:
gensim/models/word2vec_inner.c:14208:45: error: ‘PyLongObject’ {aka ‘struct _longobject’} has no member named ‘ob_digit’
14208 |     const digit* digits = ((PyLongObject*)b)->ob_digit;
      |                                             ^~
error: command '/nix/store/hgw57m0jh2w10jsv8x86ng267pm5sla8-gcc-wrapper-13.3.0/bin/gcc' failed with exit code 1
```

### Additional context

Build failed on latest master.

### Notify maintainers

- @jyp
- @mweinelt (pinging because you did a last bump)

### Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.0-113-generic, NixOS, 24.05 (Uakari), 24.05pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - channels(root): `"nixos-23.05.20230202.006c3bd"`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/cltk/default.nix
+++ b/pkgs/development/python-modules/cltk/default.nix
@@ -21,6 +21,8 @@
   stanza,
   torch,
   tqdm,
+  colorama,
+  python-dotenv,
 
   # tests
   pytestCheckHook,
@@ -28,23 +30,15 @@
 }:
 buildPythonPackage rec {
   pname = "cltk";
-  version = "1.5.0";
+  version = "2.0.4";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "cltk";
     repo = "cltk";
     tag = "v${version}";
-    hash = "sha256-aeWbfDVNn6DwW+KFh62n5RBgWp5uSWDv2RHmB27/xI4=";
+    hash = "sha256-tAomXxI6XsIAxQzPiUsT5t1CHrFDPkwyWtVuHXQCz2A=";
   };
-
-  postPatch = ''
-    substituteInPlace pyproject.toml \
-      --replace-fail "poetry>=1.1.13" poetry-core \
-      --replace-fail "poetry.masonry.api" "poetry.core.masonry.api" \
-      --replace-fail 'scipy = "<1.13.0"' 'scipy = "^1"' \
-      --replace-fail 'boltons = "^21.0.0"' 'boltons = "^24.0.0"'
-  '';
 
   build-system = [ poetry-core ];
 
@@ -67,6 +61,8 @@ buildPythonPackage rec {
     stanza
     torch
     tqdm
+    colorama
+    python-dotenv
   ];
 
   nativeCheckInputs = [

--- a/pkgs/development/python-modules/gensim/default.nix
+++ b/pkgs/development/python-modules/gensim/default.nix
@@ -1,65 +1,76 @@
 {
   lib,
   buildPythonPackage,
-  cython_0,
-  oldest-supported-numpy,
+  fetchFromGitHub,
+
+  # build-system
+  cython,
   setuptools,
-  fetchPypi,
-  mock,
+
+  # dependencies
   numpy,
   scipy,
   smart-open,
+
+  # tests
+  mock,
   pyemd,
   pytestCheckHook,
-  pythonOlder,
+  testfixtures,
 }:
 
 buildPythonPackage rec {
   pname = "gensim";
-  version = "4.3.3";
+  version = "4.4.0";
   pyproject = true;
 
-  # C code generated with CPython3.12 does not work cython_0.
-  disabled = !(pythonOlder "3.12");
-
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-hIUgdqaj2I19rFviReJMIcO4GbVl4UwbYfo+Xudtz1c=";
+  src = fetchFromGitHub {
+    owner = "piskvorky";
+    repo = "gensim";
+    tag = version;
+    hash = "sha256-TXutcU43ReBj9ss9+zBJFUxb5JqVHpl+B0c7hqcJAJY=";
   };
 
   build-system = [
-    cython_0
-    oldest-supported-numpy
+    cython
     setuptools
   ];
 
   dependencies = [
-    smart-open
     numpy
     scipy
+    smart-open
   ];
 
   nativeCheckInputs = [
     mock
     pyemd
     pytestCheckHook
-  ];
-
-  pythonRelaxDeps = [
-    "scipy"
+    testfixtures
   ];
 
   pythonImportsCheck = [ "gensim" ];
 
-  # Test setup takes several minutes
-  doCheck = false;
-
   enabledTestPaths = [ "gensim/test" ];
 
-  meta = with lib; {
+  # test_parallel is flaky under load
+  disabledTests = [ "test_parallel" ];
+
+  preCheck = ''
+    export GENSIM_DATA_DIR="$NIX_BUILD_TOP/gensim-data"
+    mkdir -p "$GENSIM_DATA_DIR"
+    export SKIP_NETWORK_TESTS=1
+  ''
+  # Prevent python from importing gensim from the source files during tests
+  + ''
+    rm gensim/__init__.py
+  '';
+
+  meta = {
     description = "Topic-modelling library";
     homepage = "https://radimrehurek.com/gensim/";
+    downloadPage = "https://github.com/piskvorky/gensim";
     changelog = "https://github.com/RaRe-Technologies/gensim/blob/${version}/CHANGELOG.md";
-    license = licenses.lgpl21Only;
+    license = lib.licenses.lgpl21Only;
   };
 }

--- a/pkgs/development/python-modules/graphrag/default.nix
+++ b/pkgs/development/python-modules/graphrag/default.nix
@@ -1,38 +1,44 @@
 {
   lib,
+  stdenv,
   buildPythonPackage,
   fetchFromGitHub,
-  poetry-core,
-  poetry-dynamic-versioning,
+
+  # build-system
+  setuptools,
+
+  # dependencies
   aiofiles,
-  aiolimiter,
+  azure-cosmos,
   azure-identity,
   azure-search-documents,
   azure-storage-blob,
-  datashaper,
   devtools,
   environs,
   fnllm,
+  future,
   graspologic,
   json-repair,
   lancedb,
-  matplotlib,
+  litellm,
   networkx,
   nltk,
   numpy,
   openai,
   pandas,
-  pyaml-env,
   pyarrow,
   pydantic,
   python-dotenv,
   pyyaml,
-  rich,
-  tenacity,
+  spacy,
+  textblob,
   tiktoken,
+  tqdm,
   typer,
   typing-extensions,
   umap-learn,
+
+  # tests
   nbformat,
   pytest-asyncio,
   pytestCheckHook,
@@ -40,54 +46,55 @@
 
 buildPythonPackage rec {
   pname = "graphrag";
-  version = "2.4.0";
+  version = "2.7.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "microsoft";
     repo = "graphrag";
-    tag = "v.${version}";
-    hash = "sha256-a8t6Nl9W/Cr7eueAvJ3dbz5G0oIhddqFMIm7HeZ8N9A=";
+    tag = "v${version}";
+    hash = "sha256-F0MiC+14KOjCVwlcZpNo15SqDOfSYsVwH8qNQTHBKPQ=";
   };
 
   build-system = [
-    poetry-core
-    poetry-dynamic-versioning
+    setuptools
   ];
 
   pythonRelaxDeps = true;
 
   dependencies = [
     aiofiles
-    aiolimiter
+    azure-cosmos
     azure-identity
     azure-search-documents
     azure-storage-blob
-    datashaper
     devtools
     environs
     fnllm
+    future
     graspologic
     json-repair
     lancedb
-    matplotlib
+    litellm
     networkx
     nltk
     numpy
     openai
     pandas
-    pyaml-env
     pyarrow
     pydantic
     python-dotenv
     pyyaml
-    rich
-    tenacity
+    spacy
+    textblob
     tiktoken
+    tqdm
     typer
     typing-extensions
     umap-learn
-  ];
+  ]
+  ++ fnllm.optional-dependencies.azure
+  ++ fnllm.optional-dependencies.openai;
 
   env.NUMBA_CACHE_DIR = "$TMPDIR";
 
@@ -101,11 +108,20 @@ buildPythonPackage rec {
 
   enabledTestPaths = [ "tests/unit" ];
 
+  disabledTestPaths = lib.optionals stdenv.hostPlatform.isDarwin [
+    # Flaky
+    "tests/unit/litellm_services/test_rate_limiter.py"
+  ];
+
   disabledTests = [
     # touch the network
+    "test_basic_functionality"
     "test_child"
     "test_dotprefix"
     "test_find"
+    "test_load_strategy_sentence"
+    "test_mixed_whitespace_handling"
+    "test_multiple_documents"
     "test_run_extract_entities_multiple_documents"
     "test_run_extract_entities_single_document"
     "test_sort_context"

--- a/pkgs/development/python-modules/graspologic/default.nix
+++ b/pkgs/development/python-modules/graspologic/default.nix
@@ -3,12 +3,15 @@
   stdenv,
   buildPythonPackage,
   fetchFromGitHub,
+
+  # build-system
   poetry-core,
   poetry-dynamic-versioning,
-  pytestCheckHook,
-  testfixtures,
+
+  # dependencies
   anytree,
   beartype,
+  future,
   gensim,
   graspologic-native,
   hyppo,
@@ -23,19 +26,33 @@
   statsmodels,
   typing-extensions,
   umap-learn,
+
+  # tests
+  pytestCheckHook,
+  testfixtures,
 }:
 
 buildPythonPackage rec {
   pname = "graspologic";
-  version = "3.4.1";
+  version = "3.4.4";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "graspologic-org";
     repo = "graspologic";
-    rev = "refs/tags/v${version}";
-    hash = "sha256-taX/4/uCQXW7yFykVHY78hJIGThEIycHwrEOZ3h1LPY=";
+    tag = "v${version}";
+    hash = "sha256-ulsb7jD/tIVEISjnNRif7VO+ZcXCAGIFl1SNZhOC7ik=";
   };
+
+  # Fix numpy 2 compat
+  postPatch = ''
+    substituteInPlace graspologic/utils/utils.py \
+      --replace-fail "np.float_" "np.float64"
+    substituteInPlace graspologic/embed/omni.py \
+      --replace-fail \
+        "A = np.array(graphs, copy=False, ndmin=3)" \
+        "A = np.asarray(graphs)"
+  '';
 
   build-system = [
     poetry-core
@@ -45,12 +62,14 @@ buildPythonPackage rec {
   pythonRelaxDeps = [
     "beartype"
     "hyppo"
+    "numpy"
     "scipy"
   ];
 
   dependencies = [
     anytree
     beartype
+    future
     gensim
     graspologic-native
     hyppo
@@ -82,19 +101,21 @@ buildPythonPackage rec {
 
   disabledTestPaths = [
     "docs"
-    "tests/test_sklearn.py"
   ]
   ++ lib.optionals stdenv.hostPlatform.isDarwin [
     # SIGABRT
     "tests/test_plot.py"
     "tests/test_plot_matrix.py"
+
+    # Hang forever
+    "tests/pipeline/embed/"
   ];
 
-  meta = with lib; {
+  meta = {
     description = "Package for graph statistical algorithms";
     homepage = "https://graspologic-org.github.io/graspologic";
-    changelog = "https://github.com/graspologic-org/graspologic/releases/tag/v${version}";
-    license = licenses.mit;
-    maintainers = with maintainers; [ bcdarwin ];
+    changelog = "https://github.com/graspologic-org/graspologic/releases/tag/${src.tag}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ bcdarwin ];
   };
 }

